crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -1173,38 +1172,33 @@ fn generate_tangents_for_mesh(mesh: &Mesh) -> Result<Vec<[f32; 4]>, GenerateTang
         other => return Err(GenerateTangentsError::UnsupportedTopology(other)),
     };
 
-    let positions = match mesh.attribute(Mesh::ATTRIBUTE_POSITION).ok_or(
-        GenerateTangentsError::MissingVertexAttribute(Mesh::ATTRIBUTE_POSITION.name),
-    )? {
-        VertexAttributeValues::Float32x3(vertices) => vertices,
-        _ => {
-            return Err(GenerateTangentsError::InvalidVertexAttributeFormat(
-                Mesh::ATTRIBUTE_POSITION.name,
-                VertexFormat::Float32x3,
-            ))
-        }
+    let VertexAttributeValues::Float32x3(positions) =
```

Comment:
same here, kind of

Reply:
While I agree for the other since matched value was quite deep in the pattern, here I think that while it introduce a bit of indentation to read the binded variable, it is still clearer than the `match` version ?

Reply:
not sure about this one, do as you like

Reply:
I guess my problem is with one overly complex statement:

```rs
let VertexAttributeValues::Float32x3(positions) =
    mesh.attribute(Mesh::ATTRIBUTE_POSITION).ok_or(
        GenerateTangentsError::MissingVertexAttribute(Mesh::ATTRIBUTE_POSITION.name),
    )?
else {
    return Err(GenerateTangentsError::InvalidVertexAttributeFormat(
        Mesh::ATTRIBUTE_POSITION.name,
        VertexFormat::Float32x3,
    ));
};
```

Breaking it into multiple simpler statements might be more readable (here, I'm creating `attribute` variable for this purpose):

```rs
let attribute = mesh.attribute(Mesh::ATTRIBUTE_POSITION).ok_or(
    GenerateTangentsError::MissingVertexAttribute(Mesh::ATTRIBUTE_POSITION.name),
)?;
let VertexAttributeValues::Float32x3(positions) = attribute else {
    return Err(GenerateTangentsError::InvalidVertexAttributeFormat(
        Mesh::ATTRIBUTE_POSITION.name,
        VertexFormat::Float32x3,
    ));
};
```

Reply:
Good suggestion, done :slightly_smiling_face: